### PR TITLE
CHE-8510: Recognize multi-module projects

### PR DESCRIPTION
**Pull request by tolusha**
### Reference issue
https://github.com/eclipse/che/issues/8510

### CODE REVIEW

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/pom/GetMavenProjectsCommand.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+      if (rootPath.isPrefixOf(project.getLocation())) {
+        projectsUri.add(ResourceUtils.fixURI(project.getLocationURI()));
+      }
+    }
+
+    return projectsUri;
+  }
+
+  private static void ensureNotCancelled(IProgressMonitor progressMonitor) {
```
Comment (svor) [inline org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/pom/GetMavenProjectsCommand.java]: You can use already existed utility method org.eclipse.che.jdt.ls.extension.core.internal.Utils#ensureNotCancelled

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/externallibrary/LibraryNavigation.java [hunk 2]
```diff
@@ -121,8 +128,39 @@
     if (javaProject == null) {
       throw new IllegalArgumentException(format("Project for '%s' not found", projectUri));
     }
+
+    IMavenProjectRegistry mavenProjectRegistry = MavenPlugin.getMavenProjectRegistry();
```
Comment (tsmaeder) [inline org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/externallibrary/LibraryNavigation.java]: I don't really understand why we would need any maven-specific stuff, here. Can you explain, please?

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/pom/GetMavenProjectsCommand.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    final String rootUri = (String) arguments.get(0);
+    final IPath rootPath = ResourceUtils.filePathFromURI(rootUri);
+    if (rootPath == null) {
+      return Collections.emptyList();
+    }
+
+    IMavenProjectRegistry mavenProjectRegistry = MavenPlugin.getMavenProjectRegistry();
+    IMavenProjectFacade[] mavenProjects = mavenProjectRegistry.getProjects();
+
+    List<String> projectsUri = new ArrayList<>(mavenProjects.length);
```
Comment (tsmaeder) [inline org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/pom/GetMavenProjectsCommand.java]: "projectUris", not "projectsUri"

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/pom/GetMavenProjectsCommand.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+   * Returns all nested maven projects starting from the given root.
+   *
+   * @param arguments the root uri
+   */
+  public static List<String> execute(List<Object> arguments, IProgressMonitor progressMonitor) {
+    Preconditions.checkArgument(!arguments.isEmpty(), "Project uri is expected.");
+    Utils.ensureNotCancelled(progressMonitor);
+
+    final String rootUri = (String) arguments.get(0);
+    final IPath rootPath = ResourceUtils.filePathFromURI(rootUri);
```
Comment (tsmaeder) [inline org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/pom/GetMavenProjectsCommand.java]: You don't need to compute this path, you can use rootUri.isPrefixOf(project.getLocatonURI)

Comment (tolusha) [reply org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/pom/GetMavenProjectsCommand.java]: `rootUri` is a String, do you mean to use `rootUri.startsWith(project.getLocation().toOSString())` ?

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/externallibrary/LibraryNavigation.java (hunk 2 above)
Comment (tolusha) [reply org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/externallibrary/LibraryNavigation.java]: There are no external libraries for for maven projects with pom packaging.
That's why we have to recognize that and return default libs to sustain current che behavior.

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/pom/GetMavenProjectsCommand.java (hunk 4 above)
Comment (tsmaeder) [reply org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/pom/GetMavenProjectsCommand.java]: yes

File: org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/externallibrary/LibraryNavigation.java (hunk 2 above)
Comment (tsmaeder) [reply org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/externallibrary/LibraryNavigation.java]: <sigh> fair enough.

**Pull request merged**